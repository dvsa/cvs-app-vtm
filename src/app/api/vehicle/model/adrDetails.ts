import { AdrDetailsAdditionalNotes } from './adrDetailsAdditionalNotes';
import { AdrDetailsApplicantDetails } from './adrDetailsApplicantDetails';
import { AdrDetailsTank } from './adrDetailsTank';
import { AdrDetailsVehicleDetails } from './adrDetailsVehicleDetails';

/**
 * Used only for HGV and TRL
 */
export interface AdrDetails { 
    vehicleDetails?: AdrDetailsVehicleDetails;
    /**
     * Optional. Applicable only if vehicleDetails.type contains the word ‘battery’.
     */
    listStatementApplicable?: boolean;
    /**
     * Mandatory if listStatementApplicable is true, therefore applicable only if vehicleDetails.type contains the word ‘battery’. Else N/A
     */
    batteryListNumber?: string;
    /**
     * Optional for all vehicle types
     */
    declarationsSeen?: boolean;
    /**
     * Optional for all vehicle types
     */
    brakeDeclarationsSeen?: boolean;
    /**
     * Optional for all vehicle types
     */
    brakeDeclarationIssuer?: string;
    /**
     * Optional for all vehicle types
     */
    brakeEndurance?: boolean;
    /**
     * Mandatory if brakeEndurance = true, else N/A
     */
    weight?: string;
    /**
     * Optional for all vehicle types
     */
    compatibilityGroupJ?: boolean;
    /**
     * Optional. Applicable only if vehicleDetails.type contains the word ‘tank’ or ‘battery’.
     */
    documents?: Array<string>;
    /**
     * Mandatory. Users must select, AT LEAST ONE of these. However, users CAN select more than one
     */
    permittedDangerousGoods?: Array<string>;
    /**
     * Optional for all vehicle types
     */
    additionalExaminerNotes?: string;
    applicantDetails?: AdrDetailsApplicantDetails;
    memosApply?: Array<string>;
    additionalNotes?: AdrDetailsAdditionalNotes;
    adrTypeApprovalNo?: string;
    adrCertificateNotes?: string;
    tank?: AdrDetailsTank;
}