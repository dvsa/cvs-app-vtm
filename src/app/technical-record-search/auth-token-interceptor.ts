import { HttpHandler, HttpInterceptor, HttpRequest, HttpEvent } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { MsAdalAngular6Service } from 'microsoft-adal-angular6';
import { mergeMap } from 'rxjs/operators';
import { Observable } from 'rxjs';

import { UserService } from './../app-user.service';

@Injectable()
export class AuthTokenInterceptor implements HttpInterceptor {
  constructor(private adal: MsAdalAngular6Service, private userSvc: UserService) {}

  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    const resource = this.adal.GetResourceForEndpoint(req.url);
    if (!resource) {
      return next.handle(req);
    }

    if (!this.adal.isAuthenticated) {
      return this.adal.acquireToken(resource).pipe(
        mergeMap((token: string) => {
          const authorizedRequest = req.clone({
            headers: req.headers.set('Authorization', `Bearer ${token}`)
          });
          return next.handle(authorizedRequest);
        })
      );
    }

    const authorizedAdalRequest = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${this.adal.accessToken}`)
    });

    this.userSvc.setUser(this.adal.userInfo.profile);
    return next.handle(authorizedAdalRequest);
  }
}
