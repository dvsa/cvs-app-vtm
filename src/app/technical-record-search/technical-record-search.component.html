<div class="govuk-width-container">
  <main class="govuk-main-wrapper">
    <div class="govuk-grid-row">
      <div class="govuk-grid-column-two-thirds">
        <h1 class="govuk-heading-xl">Search for a technical record</h1>

        <ng-container *ngIf="searchError$ | async as searchError">
          <div
            [ngClass]="{
              'govuk-form-group': true,
              'govuk-form-group--error': searchError.length > 0
            }"
          >
            <label class="govuk-label govuk-!-font-weight-bold" for="searchIdentifier">
              Vehicle registration mark, trailer ID or vehicle identification number
            </label>
            <span *ngIf="searchError.length" id="name-error" class="govuk-error-message">
              <span class="govuk-visually-hidden">Error:</span>

              <div *ngIf="searchError.includes('No resources match the search criteria.')">
                Vehicle not found, check the vehicle registration mark, trailer ID, vehicle
                identification number or change <br />
                the search criteria to find a vehicle
              </div>

              <div *ngIf="searchError.length">
                Enter a vehicle registration mark, trailer ID or vehicle identification number
              </div>

              <div
                *ngIf="
                  searchError.includes('The provided partial VIN returned more than one match.')
                "
              >
                Multiple vehicles found, search using the full vehicle identification number
              </div>
            </span>

            <input
              #searchIdentifier
              [ngClass]="{
                'govuk-input': true,
                'govuk-input--error': searchError.length > 0
              }"
              id="searchIdentifier"
              name="searchIdentifier"
              type="text"
              aria-describedby="name-error"
            />
          </div>

          <div class="govuk-form-group">
            <label class="govuk-label govuk-!-font-weight-bold">
              Search criteria
            </label>
            <select class="govuk-select" id="test-search-criteria" #searchCriteria>
              <option
                *ngFor="let searchCriteria of searchCriteriaOptions"
                [value]="searchCriteria"
                >{{ searchCriteria }}</option
              >
            </select>
          </div>

          <a
            role="button"
            (click)="searchTechRecords(searchIdentifier.value, searchCriteria.value)"
            draggable="false"
            class="govuk-button govuk-button--start govuk-!-margin-bottom-2"
            data-module="govuk-button"
          >
            Search
            <svg
              class="govuk-button__start-icon"
              xmlns="http://www.w3.org/2000/svg"
              width="17.5"
              height="19"
              viewBox="0 0 33 40"
              role="presentation"
              focusable="false"
            >
              <path fill="currentColor" d="M0 0h13l20 20-20 20H0l20-20z" />
            </svg>
          </a>
        </ng-container>
      </div>
    </div>
  </main>
</div>
