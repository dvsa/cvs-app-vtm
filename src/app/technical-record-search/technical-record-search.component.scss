@import "node_modules/govuk-frontend/govuk/all";
@import "src/app/app.component.scss";

.mt-2 {
  margin-top: 2rem;
}

.govuk-main-wrapper {

  #searchIdentifier {
    background-color: govuk-colour("white");
    border: 2px solid $govuk-input-border-colour;
    width: 589px;
    height: 40px;
  }

  .govuk-input--error {
    border: 4px solid $govuk-error-colour !important;
  }

  .govuk-error-summary {
    width: 57rem;
  }

  .govuk-error-message {
    white-space: nowrap;
    font-weight: bolder;
  }

  .govuk-error-summary {
    .govuk-error-message {
      text-decoration: underline;
    }
  }

}
