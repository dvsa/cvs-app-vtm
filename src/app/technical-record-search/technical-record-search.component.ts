import { Component, ChangeDetectionStrategy } from '@angular/core';
import { Router } from '@angular/router';
import { Store, select } from '@ngrx/store';
import { Observable } from 'rxjs';

import { IAppState } from '@app/store/state/app.state';
import { getVehicleTechRecordModelError } from '@app/store/selectors/VehicleTechRecordModel.selectors';
import { GetVehicleTechRecordModelHavingStatusAll } from '@app/store/actions/VehicleTechRecordModel.actions';
import { SEARCH_CRITERIA } from '@app/app.enums';


@Component({
  selector: 'vtm-technical-record-search',
  templateUrl: './technical-record-search.component.html',
  styleUrls: ['./technical-record-search.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TechnicalRecordSearchComponent {
  searchIdentifier = '{none searched}';
  isLoading: boolean;
  searchError$: Observable<any>;
  searchCriteriaOptions = Object.values(SEARCH_CRITERIA);
  searchCriteria = 'all';

  constructor(private router: Router, private store: Store<IAppState>) {
    this.searchError$ = this.store.pipe(select(getVehicleTechRecordModelError));
  }

  public searchTechRecords(searchIdentifier: string, searchCriteria: string) {

    this.isLoading = true;
    this.searchIdentifier = searchIdentifier;

    switch (searchCriteria) {
      case SEARCH_CRITERIA.VRM_CRITERIA:
        this.searchCriteria = 'vrm';
        break;
      case SEARCH_CRITERIA.FULL_VIN_CRITERIA:
        this.searchCriteria = 'vin';
        break;
      case SEARCH_CRITERIA.PARTIAL_VIN_CRITERIA:
        this.searchCriteria = 'partialVin';
        break;
      case SEARCH_CRITERIA.TRL_CRITERIA:
        this.searchCriteria = 'trailerId';
        break;
      case SEARCH_CRITERIA.ALL_CRITERIA:
      default:
        this.searchCriteria = 'all';
    }

    this.store.dispatch(new GetVehicleTechRecordModelHavingStatusAll([this.searchIdentifier, this.searchCriteria]));
  }
}
