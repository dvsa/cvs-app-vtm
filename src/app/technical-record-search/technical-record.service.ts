import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppConfig } from '@app/app.config';
import { Observable, of } from 'rxjs';
import { switchMap, tap, shareReplay, catchError, finalize } from 'rxjs/operators';
import { delayedRetry } from '@app/shared/delayed-retry/delayed-retry';
import { Store } from '@ngrx/store';
import { IAppState } from '@app/store/state/app.state';
import { LoadingTrue, LoadingFalse } from '@app/store/actions/Loader.actions';

@Injectable({
  providedIn: 'root'
})
export class TechnicalRecordService {
  private _apiServer = this._appConfig.settings.apiServer;
  private readonly routes;

  constructor(private httpClient: HttpClient,
    private _appConfig: AppConfig,
    private _store: Store<IAppState>) {
    this.routes = {
      // techRecords: (searchIdentifier: string) => `${this._apiServer.APITechnicalRecordServerUri}/vehicles/${searchIdentifier}/tech-records`,
      techRecordsAllStatuses: (searchIdentifier: string) => `${this._apiServer.APITechnicalRecordServerUri}/vehicles/${searchIdentifier}/tech-records?status=all&metadata=true`,
      getDocumentBlob: (vin: string) => `${this._apiServer.APIDocumentsServerUri}/vehicles/${vin}/download-file`,
      updateTechRecords: (vin: string) => `${this._apiServer.APITechnicalRecordServerUri}/vehicles/${vin}`,
    };
  }

  getTechnicalRecordsAllStatuses(searchIdentifier: string): Observable<any> {
    console.log(`getTechnicalRecordsAllStatuses url => ${this.routes.techRecordsAllStatuses(searchIdentifier)}`);
    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
    this._store.dispatch(new LoadingTrue());
    return this.httpClient.get<any[]>(this.routes.techRecordsAllStatuses(searchIdentifier), { headers }).pipe(
      delayedRetry(),
      shareReplay(),
      finalize(() => this._store.dispatch(new LoadingFalse()))
    );
  }

  updateTechnicalRecords(techRecordDto: any, vin: string): Observable<any> {
    console.log(`updateTechRecords url => ${this.routes.updateTechRecords(vin)}`);
    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
    this._store.dispatch(new LoadingTrue());
    return this.httpClient.put<any[]>(this.routes.updateTechRecords(vin), techRecordDto, { headers }).pipe(
      delayedRetry(),
      shareReplay(),
      finalize(() => this._store.dispatch(new LoadingFalse()))
    );
  }

  uploadDocuments(submitData: any): Observable<any> {
    console.log(`inside uploadDocuments received submiData => ${JSON.stringify(submitData)}`);
    return of<any>("succeeded");
  }

  getDocumentBlob(vin: string, fileName: string): Observable<{ buffer: ArrayBuffer, contentType: string, fileName?: string }> {
    console.log(`getDocumentBlob vin => ${this.routes.getDocumentBlob(vin)}`);
    this._store.dispatch(new LoadingTrue());
    return this.httpClient.get<
      {
        fileBuffer: {
          type: string,
          data: Array<number>
        },
        contentType: string
      }>(this.routes.getDocumentBlob(vin), { params: { filename: fileName }, responseType: 'json' }).pipe(
        switchMap(response => {
          const ab = new ArrayBuffer(response.fileBuffer.data.length);
          const view = new Uint8Array(ab);
          for (let i = 0; i < response.fileBuffer.data.length; i++) {
            view[i] = response.fileBuffer.data[i];
          }
          return of({ buffer: ab, contentType: response.contentType, fileName: fileName });
        }),
        tap(_ => console.log(`getDocumentBlob => ${JSON.stringify(_)}`)),
        delayedRetry(),
        shareReplay(),
        finalize(() => this._store.dispatch(new LoadingFalse()))
      );
  }
}
