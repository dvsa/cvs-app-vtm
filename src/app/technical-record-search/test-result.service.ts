import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {MsAdalAngular6Service} from 'microsoft-adal-angular6';
import { environment } from '@environments/environment';

const routes = {
  testResults: (searchIdentifier: string) => `${environment.APITestResultServerUri}/test-results/${searchIdentifier}`,
};


@Injectable({
  providedIn: 'root'
})

export class TestResultService {
  private jwttoken: string;

  constructor(private httpClient: HttpClient, private adalSvc: MsAdalAngular6Service) {
  }

  getTestResults(searchIdentifier: string): Observable<any> {
    // tslint:disable-next-line:max-line-length
    const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8').set('Authorization', `Bearer ${this.jwttoken}`);
    return this.httpClient.get<any[]>(routes.testResults(searchIdentifier), {headers});
  }
}
