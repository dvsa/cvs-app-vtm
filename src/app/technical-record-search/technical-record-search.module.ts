import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {HttpClientModule} from '@angular/common/http';
import {FormsModule, ReactiveFormsModule} from '@angular/forms';
import {FontAwesomeModule} from '@fortawesome/angular-fontawesome';
import {MatFormFieldModule} from '@angular/material/form-field';
import {SharedModule} from '../shared/shared.module';
import {NgrxFormsModule} from 'ngrx-forms';
import {NgxJsonViewerModule} from 'ngx-json-viewer';
import {MaterialModule} from '@app/material.module';
import {TechnicalRecordSearchComponent} from './technical-record-search.component';
import {RouterModule} from '@angular/router';
import {AuthenticationGuard} from 'microsoft-adal-angular6';
import {TechnicalRecordService} from './technical-record.service';
import {TestResultService} from './test-result.service';
import {AdrReasonModalComponent} from '@app/shared/adr-reason-modal/adr-reason-modal.component';
import { EffectsModule } from '@ngrx/effects';
import { VehicleTestResultModelEffects } from '@app/store/effects/VehicleTestResultModel.effects';
import { VehicleTechRecordModelEffects } from '@app/store/effects/VehicleTechRecordModel.effects';

@NgModule({
  imports: [
    CommonModule,
    MaterialModule,
    MatFormFieldModule,
    HttpClientModule,
    RouterModule.forChild([
      {path: '', component: TechnicalRecordSearchComponent, canActivate: [AuthenticationGuard]},

    ]),
    EffectsModule.forFeature([VehicleTechRecordModelEffects, VehicleTestResultModelEffects]),
    FormsModule,
    SharedModule,
    FontAwesomeModule,
    ReactiveFormsModule,
    NgrxFormsModule,
    NgxJsonViewerModule,
  ],
  declarations: [
    TechnicalRecordSearchComponent,
  ],
  exports: [
    TechnicalRecordSearchComponent,
  ],
  providers: [TechnicalRecordService, TestResultService],
  entryComponents: [AdrReasonModalComponent],
})
export class TechnicalRecordSearchModule {
}
