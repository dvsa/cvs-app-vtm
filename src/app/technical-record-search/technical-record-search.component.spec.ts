import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { INITIAL_STATE, Store, StoreModule } from '@ngrx/store';

import { hot } from 'jasmine-marbles';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { adrDetailsReducer } from '@app/technical-record/adr-details/adr-details-form/store/adrDetails.reducer';
import { appReducers } from '@app/store/reducers/app.reducers';
import { IAppState } from '@app/store/state/app.state';
import { TechnicalRecordSearchComponent } from './technical-record-search.component';
import {GetVehicleTechRecordModelHavingStatusAll} from '@app/store/actions/VehicleTechRecordModel.actions';

describe('TechnicalRecordSearchComponent', () => {

  let component: TechnicalRecordSearchComponent;
  let fixture: ComponentFixture<TechnicalRecordSearchComponent>;
  let store: Store<IAppState>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [TechnicalRecordSearchComponent],
      imports: [
        RouterTestingModule,
        StoreModule.forRoot(appReducers),
        StoreModule.forFeature('adrDetails', adrDetailsReducer),
      ],
      providers: [
        {
          provide: Store,
          useValue: {
            dispatch: jest.fn(),
            pipe: jest.fn(() => hot('-a', { a: INITIAL_STATE })),
            select: jest.fn()
          }
        },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    })
      .compileComponents();
  });

  beforeEach(() => {
    store = TestBed.get(Store);
    spyOn(store, 'dispatch').and.callThrough();
    fixture = TestBed.createComponent(TechnicalRecordSearchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  afterEach(() => {
    fixture.destroy();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    //expect(fixture).toMatchSnapshot();
  });

  describe('searchTechRecords', () => {
    it('should dispatch get action when the button is pressed', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'Vehicle registration mark, trailer ID or vehicle identification number';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.isLoading).toBe(true);
      expect(component.searchIdentifier).toBe(searchIdentifier);
      expect(component.searchCriteria).toBe('all');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

    it('should have correct behaviour for VRM', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'Vehicle registration mark (VRM)';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.searchCriteria).toBe('vrm');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

    it('should have correct behaviour for FULL VIN', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'Full vehicle identification number (VIN)';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.searchCriteria).toBe('vin');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

    it('should have correct behaviour for PARTIAL VIN', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'Partial VIN (last 6 characters)';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.searchCriteria).toBe('partialVin');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

    it('should have correct behaviour for TRAILER ID', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'Trailer ID';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.searchCriteria).toBe('trailerId');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

    it('should have correct behaviour for default', () => {
      const searchIdentifier = 'test';
      const searchCriteria   = 'testing';
      component.searchTechRecords(searchIdentifier, searchCriteria);

      expect(component.searchCriteria).toBe('all');

      expect(store.dispatch).toHaveBeenCalledWith(new GetVehicleTechRecordModelHavingStatusAll([component.searchIdentifier, component.searchCriteria]));
    });

  });
});
