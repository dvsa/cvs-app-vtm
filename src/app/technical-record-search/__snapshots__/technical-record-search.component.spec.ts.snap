// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnicalRecordSearchComponent should create 1`] = `
<vtm-technical-record-search
  router={[Function Router]}
  searchError$={[Function Store]}
  searchIdentifier={[Function String]}
  store={[Function Store]}
>
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
    >
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <h1
            class="govuk-heading-xl"
          >
            Search for a technical record
          </h1>
          <div
            class="govuk-form-group"
            ng-reflect-ng-class="[object Object]"
          >
            <label
              class="govuk-label"
              for="searchIdentifier"
            >
               Vehicle registration mark, trailer ID or vehicle identification number 
            </label>
            
            <input
              aria-describedby="name-error"
              class="govuk-input"
              id="searchIdentifier"
              name="searchIdentifier"
              ng-reflect-ng-class="[object Object]"
              type="text"
            />
          </div>
          <a
            class="govuk-button govuk-button--start mt-1"
            data-module="govuk-button"
            draggable="false"
            role="button"
          >
             Search 
            <svg
              class="govuk-button__start-icon"
              focusable="false"
              height="19"
              role="presentation"
              viewBox="0 0 33 40"
              width="17.5"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h13l20 20-20 20H0l20-20z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </main>
  </div>
</vtm-technical-record-search>
`;
